sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -80,10 +80,21 @@ private[hive] object IsolatedClientLoader extends Logging {
       resolvedVersions((resolvedVersion, actualHadoopVersion))
     }
 
+    // HiveMetastoreConnection using maven missed the jars in classpath which passed
+    // through --jars. They need to be added back before new IsolatedClientLoader()
+    val jarsInClasspath = Utils.getContextOrSparkClassLoader match {
+      case m: MutableURLClassLoader =>
+        val jars = m.getURLs.toSeq
+        logInfo(s"Added jars in classpath to IsolatedClientLoader: $jars")
+        jars
+      case _ =>
+        Seq.empty
+    }
+
     new IsolatedClientLoader(
       hiveVersion(hiveMetastoreVersion),
       sparkConf,
-      execJars = files,
+      execJars = files ++ jarsInClasspath,
```

Comment:
I have some small point:

1. jar passed by --jars or `spark.jars` will be set to `SparkContext.getActive.get.listJars()`
2. mabe we need consider java version problem as https://github.com/apache/spark/blob/9f3c82163a52c3d4cef9580d4c4a692754163281/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala#L350
3. The Third way to create IsolatedClientLoader  will have same problem. https://github.com/apache/spark/blob/9f3c82163a52c3d4cef9580d4c4a692754163281/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala#L391

Reply:
Thank you @AngersZhuuuu 
1. `SparkContext.getActive.get.jars` will return empty list in the YARN mode.
2. I think checking JDK is no need since the initialization process of MutableURLClassLoader had handled it. BTW I have tested in JDK12.
```scala
case m: MutableURLClassLoader => ...
case _ => Seq.empty
```

Reply:
That's good, thanks.